pkg/agent/multicluster/streched_networkpolicy_controller.go

```diff
@@ -0,0 +1,375 @@
+func (s *StretchedNetworkPolicyController) processPodUpdate(old, cur interface{}) {
+	oldPod, oldIsPod := old.(*v1.Pod)
+	curPod, curIsPod := cur.(*v1.Pod)
+	if !oldIsPod || !curIsPod {
+		klog.V(2).InfoS("StretchedNetworkPolicyController Pod UpdateFunc received unexpected object", "oldObj", old, "curObj", cur)
+		return
+	}
+	if oldPod != nil && reflect.DeepEqual(oldPod.Labels, curPod.Labels) {
+		klog.V(2).InfoS("StretchedNetworkPolicyController Pod UpdateFunc received a Pod update event, but labels are the same. Skip it.", "podName", curPod.Name, "podNamespace", curPod.Namespace)
+		return
+	}
+	s.enqueuePod(curPod)
+}
+
+// processPodDelete handles Pod delete events. It deletes the Pod from the labelToPods and podToLabel.
+func (s *StretchedNetworkPolicyController) processPodDelete(old interface{}) {
+	oldPod, oldIsPod := old.(*v1.Pod)
+	if !oldIsPod {
+		klog.V(2).InfoS("StretchedNetworkPolicyController Pod DeleteFunc received unexpected object", "oldObj", old)
+		return
+	}
+	oldPodRef := s.getPodReference(oldPod)
+	podLabel := s.podToLabel[oldPodRef]
+	s.deleteLabelToPod(podLabel, oldPod)
+	s.deletePodToLabel(oldPod)
+}
+
+// processNamespaceUpdate handles Namespace update events. It only enqueue all Pods in this Namespace if the Labels of this Namespace has been updated.
+func (s *StretchedNetworkPolicyController) processNamespaceUpdate(old, cur interface{}) {
+	oldNS, oldIsNS := old.(*v1.Namespace)
+	curNS, curIsNS := cur.(*v1.Namespace)
+	if !oldIsNS || !curIsNS {
+		klog.V(2).InfoS("StretchedNetworkPolicyController Namespace UpdateFunc received unexpected object", "oldObj", old, "curObj", cur)
+		return
+	}
+	if reflect.DeepEqual(oldNS.Labels, curNS.Labels) {
+		klog.V(2).InfoS("StretchedNetworkPolicyController Namespace UpdateFunc received a Namespace update event, but labels are the same. Skip it.", "Namespace", curNS.Name)
+		return
+	}
+	allPodsInNS, err := s.podLister.Pods(curNS.Name).List(labels.Everything())
+	if err != nil {
+		klog.ErrorS(err, "can't list Pods from Namespace", "Namespace", curNS.Name)
+	}
+	for _, pod := range allPodsInNS {
+		s.enqueuePod(pod)
+	}
+}
+
+// processLabelIdentityAddOrUpdate handles labelIdentity add or update event. It will
+// 1. Add/Update labelIdentity to labelIdentityCache
+// 2. Enqueue all pods with this label Identity
+func (s *StretchedNetworkPolicyController) processLabelIdentityAddOrUpdate(obj interface{}) {
+	labelIdentity, isLabelID := obj.(*v1alpha1.LabelIdentity)
+	if !isLabelID {
+		klog.V(2).InfoS("StretchedNetworkPolicyController LabelIdentity Add/UpdateFunc received unexpected object", "obj", obj)
+		return
+	}
+	s.labelIdentityCache[labelIdentity.Spec.Label] = labelIdentity.Spec.ID
+	if podSet, ok := s.labelToPods[labelIdentity.Spec.Label]; ok {
+		for podRef := range podSet {
+			if pod, err := s.podLister.Pods(podRef.namespace).Get(podRef.name); err == nil {
+				s.enqueuePod(pod)
```

Comment:
I feel `s.enqueuePod` is no needed since you are passing Pod to it. Could you check other occurrences?

Reply:
Good one. Done.